go/vt/sqlparser/ast.go

```diff
@@ -2640,6 +2631,109 @@ type (
 		JSONValue Expr
 	}
 
+	AggrFunc interface {
+		Expr
+		AggrName() string
+		GetArg() Expr
+		isDistinct() bool
+		GetArgs() Exprs
+	}
+
+	Count struct {
+		Args     Exprs
+		Distinct bool
+		Name     string
+	}
+
+	CountStar struct {
+		Distinct bool
+		Name     string
+	}
+
+	Avg struct {
+		Args     Exprs
```

Comment:
for `avg`, and the rest of the aggr functions except `count`, a single argument is all we need to store